test/fuzz/tests/rpc_jsonrpc_server_test.go

```diff
@@ -0,0 +1,69 @@
+//go:build gofuzz || go1.18
+
+package tests
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/tendermint/tendermint/libs/log"
+	rs "github.com/tendermint/tendermint/rpc/jsonrpc/server"
+	"github.com/tendermint/tendermint/rpc/jsonrpc/types"
+)
+
+var rpcFuncMap = map[string]*rs.RPCFunc{
+	"c": rs.NewRPCFunc(func(ctx context.Context, s string, i int) (string, error) {
+		return "foo", nil
+	}, "s", "i"),
+}
+var mux *http.ServeMux
+
+func init() {
+	mux = http.NewServeMux()
+	rs.RegisterRPCFuncs(mux, rpcFuncMap, log.NewNopLogger())
+}
+
+func FuzzRPCJSONRPCServer(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		if len(data) == 0 {
+			return
+		}
+
+		req, _ := http.NewRequest("POST", "http://localhost/", bytes.NewReader(data))
```

Comment:
This should fail loudly on error (since the error is never expected to occur, but could if the test gets modified).

```suggestion
		req, err := http.NewRequest("POST", "http://localhost/", bytes.NewReader(data))
		if err != nil {
			panic(err)
		}
```

Reply:
Done.